test/functional/android/touch_action_tests.py

```diff
@@ -135,104 +126,105 @@ def test_press_and_moveto_x_y(self):
         self.assertIsNotNone(el)
 
     def test_drag_and_drop(self):
-        el1 = self.driver.find_element_by_name('Content')
-        el2 = self.driver.find_element_by_name('Animation')
+        el1 = self.driver.find_element_by_accessibility_id('Content')
+        el2 = self.driver.find_element_by_accessibility_id('Animation')
         self.driver.scroll(el1, el2)
 
-        el = self.driver.find_element_by_name('Views')
+        el = self.driver.find_element_by_accessibility_id('Views')
         action = TouchAction(self.driver)
         action.tap(el).perform()
 
-        el = self.driver.find_element_by_name('Drag and Drop')
+        el = wait_for_element(self.driver, MobileBy.ACCESSIBILITY_ID, 'Drag and Drop', SLEEPY_TIME)
         action.tap(el).perform()
 
-        dd3 = self.driver.find_element_by_id('com.example.android.apis:id/drag_dot_3')
+        dd3 = wait_for_element(self.driver, MobileBy.ID, 'com.example.android.apis:id/drag_dot_3', SLEEPY_TIME)
         dd2 = self.driver.find_element_by_id('com.example.android.apis:id/drag_dot_2')
 
         # dnd is stimulated by longpress-move_to-release
         action.long_press(dd3).move_to(dd2).release().perform()
 
-        el = self.driver.find_element_by_id('com.example.android.apis:id/drag_result_text')
-        self.assertEqual('Dropped!', el.get_attribute('text'))
+        el = self.driver.find_element_by_id('com.example.android.apis:id/drag_text')
+        self.assertTrue('drag_dot_3' in el.text)
 
     def test_driver_drag_and_drop(self):
-        el1 = self.driver.find_element_by_name('Content')
-        el2 = self.driver.find_element_by_name('Animation')
+        el1 = self.driver.find_element_by_accessibility_id('Content')
+        el2 = self.driver.find_element_by_accessibility_id('Animation')
         self.driver.scroll(el1, el2)
 
-        el = self.driver.find_element_by_name('Views')
+        el = self.driver.find_element_by_accessibility_id('Views')
         action = TouchAction(self.driver)
         action.tap(el).perform()
 
-        el = self.driver.find_element_by_name('Drag and Drop')
+        el = wait_for_element(self.driver, MobileBy.ACCESSIBILITY_ID, 'Drag and Drop', SLEEPY_TIME)
         action.tap(el).perform()
 
         dd3 = self.driver.find_element_by_id('com.example.android.apis:id/drag_dot_3')
         dd2 = self.driver.find_element_by_id('com.example.android.apis:id/drag_dot_2')
 
         self.driver.drag_and_drop(dd3, dd2)
 
-        el = self.driver.find_element_by_id('com.example.android.apis:id/drag_result_text')
-        self.assertEqual('Dropped!', el.get_attribute('text'))
+        el = self.driver.find_element_by_id('com.example.android.apis:id/drag_text')
+        self.assertTrue('drag_dot_3' in el.text)
 
     def test_driver_swipe(self):
-        self.assertRaises(NoSuchElementException, self.driver.find_element_by_name, 'Views')
+        el = self.driver.find_element_by_accessibility_id('Views')
```

Comment:
[notice] 'Views' already shows without scrolling dow, so moved to 'Views' and checked 'ImageView'